### PR TITLE
043create conn location internship

### PR DESCRIPTION
**Pull request by OlgaEvdokimova**
Create connection internship-location
Edit FormRegisterDTO by CountryDTO and CityDTO

### CODE REVIEW

**Review by Maksim-maevsky — DISMISSED** [review]

**Review by Fenydude — DISMISSED** [review]

File: src/main/java/by/exadel/internship/dto/formDTO/FormRegisterDTO.java [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 import by.exadel.internship.dto.TimeForCallDTO;
 import by.exadel.internship.dto.enums.EnglishLevel;
+import by.exadel.internship.dto.locationDTO.CityDTO;
+import by.exadel.internship.dto.locationDTO.CountryDTO;
 import by.exadel.internship.entity.location.City;
```
Comment (mortal-gpt) [inline src/main/java/by/exadel/internship/dto/formDTO/FormRegisterDTO.java]: extra import of City and Country ?

File: src/main/java/by/exadel/internship/dto/internshipDTO/GuestInternshipDTO.java [hunk 2]
```diff
@@ -40,7 +40,7 @@
     private String techSkills;
 
     @NotNull
-    private List<LocationDTO> countryList;
+    private List<LocationDTO> locationListDTO;
```
Comment (mortal-gpt) [inline src/main/java/by/exadel/internship/dto/internshipDTO/GuestInternshipDTO.java]:
```suggestion
    private List<LocationDTO> locations;
```

or 

```suggestion
    private List<LocationDTO> locationList;
```

File: src/test/java/by/exadel/internship/formTest/FormTest.java [hunk 3]
```diff
@@ -19,7 +21,9 @@
 
 
 public class FormTest extends InternshipApplicationTests {
-
+    @Test
```
Comment (mortal-gpt) [inline src/test/java/by/exadel/internship/formTest/FormTest.java]: Do we need this empty test?

Comment (OlgaEvdokimova) [reply src/test/java/by/exadel/internship/formTest/FormTest.java]: No, It was for test,  dropped it

File: src/main/java/by/exadel/internship/dto/formDTO/FormRegisterDTO.java (hunk 1 above)
Comment (OlgaEvdokimova) [reply src/main/java/by/exadel/internship/dto/formDTO/FormRegisterDTO.java]: yes, thanks, dropped

File: src/main/java/by/exadel/internship/dto/internshipDTO/GuestInternshipDTO.java (hunk 2 above)
Comment (OlgaEvdokimova) [reply src/main/java/by/exadel/internship/dto/internshipDTO/GuestInternshipDTO.java]: done

Comment (OlgaEvdokimova) [reply src/main/java/by/exadel/internship/dto/internshipDTO/GuestInternshipDTO.java]: image

**Review by Fenydude — DISMISSED** [review]

File: files/052a9128-3e56-4ab7-9ae2-0f5f8039cab3/text.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+some file
```
Comment (mortal-gpt) [inline files/052a9128-3e56-4ab7-9ae2-0f5f8039cab3/text.txt]: useless file?

File: files/59f066a0-b859-4cdb-a1a5-6a14fabcb1e5/text.txt [hunk 5]
```diff
@@ -0,0 +1 @@
+some file
```
Comment (mortal-gpt) [inline files/59f066a0-b859-4cdb-a1a5-6a14fabcb1e5/text.txt]: useless file?

File: src/test/java/by/exadel/internship/formTest/FormTest.java [hunk 6]
```diff
@@ -60,15 +62,18 @@ public void givenFormWithFile_checkTestData()
         assertEquals(formFullDTO.getSkype(), "string");
     }
 
-    @Test
-    public void checkListSize() throws Exception {
```
Comment (mortal-gpt) [inline src/test/java/by/exadel/internship/formTest/FormTest.java:64]: What's wrong with test?

File: src/main/java/by/exadel/internship/dto/locationDTO/CityDTO.java [hunk 7]
```diff
@@ -7,11 +7,9 @@
 
 import java.util.UUID;
 @Data
-@Builder
 @NoArgsConstructor
 @AllArgsConstructor
 public class CityDTO {
 
-    private UUID id;
```
Comment (kzaretskaya) [inline src/main/java/by/exadel/internship/dto/locationDTO/CityDTO.java]: Why did you remove id?  I think it'll be useful to have it

File: src/main/java/by/exadel/internship/dto/formDTO/FormRegisterDTO.java [hunk 8]
```diff
@@ -2,14 +2,16 @@
 
 import by.exadel.internship.dto.TimeForCallDTO;
 import by.exadel.internship.dto.enums.EnglishLevel;
-import by.exadel.internship.entity.location.City;
-import by.exadel.internship.entity.location.Country;
+import by.exadel.internship.dto.locationDTO.CityDTO;
```
Comment (kzaretskaya) [inline src/main/java/by/exadel/internship/dto/formDTO/FormRegisterDTO.java]: remove please from all package name `DTO`
```suggestion
import by.exadel.internship.dto.location.CityDTO;
```

File: src/main/resources/application.yaml [hunk 9]
```diff
@@ -24,6 +24,9 @@ spring:
       - 'classpath:db/changelog/v.0.4/sql-import/import-user-internship.sql'
       - 'classpath:db/changelog/v.0.4/sql-import/import-user-hash-password.sql'
       - 'classpath:db/changelog/v.0.3/sql-import/import-time-for-call.sql'
+      - 'classpath:db/changelog/v.0.5/sql-import/location.sql'
```
Comment (SergeyKoval) [inline src/main/resources/application.yaml:27]: strange order of changes: 

1. v.0.4
2. v.0.3
3. v.0.5

**Review by SergeyKoval — CHANGES_REQUESTED** [review]

File: src/test/java/by/exadel/internship/formTest/FormTest.java (hunk 6 above)
Comment (OlgaEvdokimova) [reply src/test/java/by/exadel/internship/formTest/FormTest.java:64]: We changed endpoint(appended parametr internshipId) It will be redone soon

File: files/59f066a0-b859-4cdb-a1a5-6a14fabcb1e5/text.txt (hunk 5 above)
Comment (OlgaEvdokimova) [reply files/59f066a0-b859-4cdb-a1a5-6a14fabcb1e5/text.txt]: Forgot to drop before commit, after testing saving form. Thanks

File: files/052a9128-3e56-4ab7-9ae2-0f5f8039cab3/text.txt (hunk 4 above)
Comment (OlgaEvdokimova) [reply files/052a9128-3e56-4ab7-9ae2-0f5f8039cab3/text.txt]: Forgot to drop before commit, after testing saving form. Thanks

File: src/main/java/by/exadel/internship/service/impl/FormServiceImpl.java [hunk 10]
```diff
@@ -62,7 +68,14 @@ public FormFullDTO process(FormRegisterDTO form, MultipartFile file) {
 
     private FormFullDTO saveForm(FormRegisterDTO formRegisterDTO) {
 
+        Country country = countryRepository.findByName(formRegisterDTO.getCountry().getName());
```
Comment (kzaretskaya) [inline src/main/java/by/exadel/internship/service/impl/FormServiceImpl.java]: If you try to find by name it should cover exception cases (like Not found).
Question: Why can't UI send country/city id?

File: src/main/resources/db/changelog/v.0.5/db.changelog-table-location.yaml [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+              - column:
+                  name: l_county_id
+                  type: UUID
+                  constraints:
+                    nullable: false
+              - column:
+                  name: l_city_id
+                  type: UUID
+                  constraints:
+                    nullable: false
```
Comment (kzaretskaya) [inline src/main/resources/db/changelog/v.0.5/db.changelog-table-location.yaml]: probably it can be null since internship can be applied on whole country, must not particular city.

File: src/main/java/by/exadel/internship/dto/formDTO/FormRegisterDTO.java (hunk 8 above)
Comment (OlgaEvdokimova) [reply src/main/java/by/exadel/internship/dto/formDTO/FormRegisterDTO.java]: Done

File: src/main/resources/db/changelog/v.0.5/db.changelog-table-location.yaml (hunk 11 above)
Comment (OlgaEvdokimova) [reply src/main/resources/db/changelog/v.0.5/db.changelog-table-location.yaml]: Done

File: src/main/java/by/exadel/internship/service/impl/FormServiceImpl.java (hunk 10 above)
Comment (OlgaEvdokimova) [reply src/main/java/by/exadel/internship/service/impl/FormServiceImpl.java]: Done,thanks
Question: Why can't UI send country/city id? -> they can, and the will. I talked with them.

File: src/main/java/by/exadel/internship/dto/locationDTO/CityDTO.java (hunk 7 above)
Comment (OlgaEvdokimova) [reply src/main/java/by/exadel/internship/dto/locationDTO/CityDTO.java]: I got you. Done

File: src/main/java/by/exadel/internship/controller/LocationController.java [hunk 12]
```diff
@@ -28,7 +29,7 @@
 
     @GetMapping("/cities")
     @ApiOperation("return city list")
-    public List<CityDTO> getCityList(@RequestParam(value = "name") String countryName){
-        return locationService.getCitiesByCountryName(countryName);
+    public List<CityDTO> getCityList(@RequestParam(value = "id") UUID countryId){
```
Comment (kzaretskaya) [inline src/main/java/by/exadel/internship/controller/LocationController.java]:
```suggestion
    public List<CityDTO> getCityList(@RequestParam(value = "countryId") UUID countryId){
```

File: src/main/resources/application.yaml (hunk 9 above)
Comment (OlgaEvdokimova) [reply src/main/resources/application.yaml:27]: Can I Leave now this variant ? Because I need time to think the orders appended columns to order them ideally 
image

File: src/main/java/by/exadel/internship/controller/LocationController.java (hunk 12 above)
Comment (OlgaEvdokimova) [reply src/main/java/by/exadel/internship/controller/LocationController.java]: Done

**Review by kzaretskaya — APPROVED** [review]

File: src/main/resources/application.yaml (hunk 9 above)
Comment (SergeyKoval) [reply src/main/resources/application.yaml:27]: Correct please in next PR

**Review by SergeyKoval — APPROVED** [review]
